apps/glusterfs/brick_create.go

```diff
@@ -34,11 +34,13 @@ func createDestroyConcurrently(db *bolt.DB,
 		sg.Add(1)
 		go func(b *BrickEntry) {
 			defer sg.Done()
+			sg.Lock()
```

Comment:
This is not good, because this essentially would make Heketi take a long time on large number of parallel bricks. Instead, review how to use the existing infrastructure from the state change call.  You can also use "db" key locks in the way the Register/Unregister work.  For example, if you are "deleting" a brick, you cannot change its state until the state change has fully happened.  You can use the db to place it in "XXXING" state until it gets to "XXX" state.